src/Faithlife.Json/ContractResolvers/SnakeCasePropertyNamesContractResolver.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json.Serialization;
+
+namespace Faithlife.Json.ContractResolvers
+{
+	public sealed class SnakeCasePropertyNamesContractResolver : DefaultContractResolver
+	{
+		protected override string ResolvePropertyName(string propertyName)
+		{
+			return s_snakeCaseRegex.Replace(propertyName,
+				match => (match.Groups[1].Success ? match.Groups[1] + "_" : "") + match.Groups[2].ToString().ToLowerInvariant());
+		}
+
+		private static readonly Regex s_snakeCaseRegex = new Regex(@"([a-z])?([A-Z]+)");
```

Comment:
Should we handle numbers?

FWIW, here's how Facility does [snake case](https://github.com/FacilityApi/Facility/blob/5d5073737726323b18dc517feccd2a65b9aea40c/src/Facility.Definition/CodeGen/CodeGenUtility.cs#L53-L64) ([tests](https://github.com/FacilityApi/Facility/blob/63c8a822a9d8458fffe12d3b222461631a45fd2b/tests/Facility.Definition.UnitTests/CodeGen/CodeGenUtilityTests.cs#L47-L56)). It doesn't merge acronyms into the following word.